src/parsers/mimic.py

```diff
@@ -0,0 +1,233 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import os
+
+from src.utils.utils import AnalysisUtils, change_col_to_datetime
+from src.utils.constants import LAB_MAPPING, LAB_VECT_COLS
+
+
+class MIMICParser(AnalysisUtils):
+    def __init__(self, data, res, gender="MF", age_b=0, age_a=100, ethnicity="WHITE", lab_mapping=None):
+        self.m_p_df = None
+        AnalysisUtils.__init__(self, data, res, gender=gender, age_b=age_b, age_a=age_a, ethnicity=ethnicity, lab_mapping=LAB_MAPPING)
+        
+    def generate_med_data(self):
+        admits = pd.read_csv(os.path.join(self.data, "mimiciii/1.4/raw", "ADMISSIONS.csv.gz"))
+        inputevents_mv = pd.read_csv(os.path.join(self.data, "mimiciii/1.4/raw", "INPUTEVENTS_MV.csv.gz"))
+                
+        ### Merge medication and admission data
+        inputevents_mv = pd.merge(inputevents_mv, admits, how="inner", on=["HADM_ID", "SUBJECT_ID"])
+        inputevents_mv = change_col_to_datetime(inputevents_mv, 'ADMITTIME')
+        inputevents_mv = change_col_to_datetime(inputevents_mv, 'ENDTIME')
+        inputevents_mv = change_col_to_datetime(inputevents_mv, 'STARTTIME')
+        inputevents_mv['MedTimeFromAdmit'] = inputevents_mv['ENDTIME']-inputevents_mv['ADMITTIME']
```

Comment:
Why do we need to do that here?
`dt.total_seconds()/3600` converts the `datetime` data to `hours`, thus reducing the precision of the data. I feel its better if we persist the `datetime` data so that we can choose precision based on use case in a later stage.